apps/browser/src/autofill/services/autofill.service.ts

```diff
@@ -260,22 +256,29 @@ export default class AutofillService implements AutofillServiceInterface {
   }
 
   /**
-   * Autofills the active tab with the next login item from the cache
-   * @param pageDetails The data scraped from the page
-   * @param fromCommand Whether the autofill is triggered by a keyboard shortcut (`true`) or autofill on page load (`false`)
-   * @returns The TOTP code of the successfully autofilled login, if any
+   * Autofill the active tab with the next login item from the cache
+   * @param {PageDetail[]} pageDetails The data scraped from the page
+   * @param {boolean} fromCommand Whether the autofill is triggered by a keyboard shortcut (`true`) or autofill on page load (`false`)
+   * @returns {Promise<string | null>} The TOTP code of the successfully autofilled login, if any
    */
-  async doAutoFillActiveTab(pageDetails: PageDetail[], fromCommand: boolean): Promise<string> {
+  async doAutoFillActiveTab(
+    pageDetails: PageDetail[],
+    fromCommand: boolean
+  ): Promise<string | null> {
```

Comment:
With the early return, should this be

```suggestion
  ): Promise<string | null> | null {
```

Reply:
Hmm... not quite, if a method is delineated as an `async` method, then all values returned from the method are wrapped in a Promise, this includes void return values. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#description 

As a result, this method will return a Promise with a potential `null` value, even if the `await`-ed method is not hit.

Reply:
ah right - I totally missed the `async` 😅